### PR TITLE
Vk fix client todos

### PR DESCRIPTION
**Pull request by cubisttriangle**

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -300,3 +300,62 @@ pub struct Metric {
...
+            MarketNewsCategory::General => "general",
+            MarketNewsCategory::Forex => "forex",
+            MarketNewsCategory::Crypto => "crypto",
+            MarketNewsCategory::Merger => "merger",
+        })?;
+        Ok(())
+    }
+}
+#[derive(Debug)]
+pub enum Profile2Param {
```
Comment (henryboisdequin) [inline src/types.rs]:
```suggestion
pub enum ProfileToParam {
```

Small nit :)

File: src/types.rs [hunk 2]
```diff
@@ -300,3 +300,62 @@ pub struct Metric {
     pub totalDebtCagr5Y: Option<f64>,
     pub yearToDatePriceReturnDaily: Option<f64>,
 }
+
+// TODO: Is there a macro we could make to automate From/Display traits?
```
Comment (henryboisdequin) [inline src/types.rs:304]: This would have to be a proc macro if we wanted to do something like this, I mean this would be a whole nother crate

**Review by henryboisdequin — CHANGES_REQUESTED** [review]
Thanks for this, looks great! I just have some small nits and a comment.

**Comment by cubisttriangle** [comment]
I checked this run's tests and last run's tests and it seems like every test that relies on the network fails. I wonder what could be wrong?

File: src/types.rs (hunk 2 above)
Comment (cubisttriangle) [reply src/types.rs:304]: Ahh, damn. Well maybe we can figure something out if it becomes a pain.

File: src/types.rs (hunk 1 above)
Comment (cubisttriangle) [reply src/types.rs]: fixed

File: src/types.rs (hunk 2 above)
Comment (henryboisdequin) [reply src/types.rs:304]: Yeah, I mean it does become painful for any application so I will consider building such a crate

It needs to be a proc macro as we are accessing the fields of the struct and looping over them and generating a `impl Display/From` which would require looking at the stream of tokens.

**Comment by henryboisdequin** [comment]
> I checked this run's tests and last run's tests and it seems like every test that relies on the network fails. I wonder what could be wrong?

Same, I'll look into it. Otherwise, thanks for this PR!

**Comment by henryboisdequin** [comment]
Also, squash + merge is now the default :)

**Comment by cubisttriangle** [comment]
Looks like a doc-test also failed, maybe you can fix it with less hassle:

```
diff --git a/src/lib.rs b/src/lib.rs
index d9a5264..08d6b27 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -17,7 +17,7 @@
 //!     // Create a new finnhub client.
 //!     let client = Client::new(api_key);
 //!     // Get a list of supported stocks given the exchange.
-//!     let res = client.stock_symbol("US".into()).await.unwrap();
+//!     let res = client.stock_symbol("US".into(), None, None, None).await.unwrap();
 //!     // Print out the results.
 //!     println!("{:#?}", res);
 //!}
```

File: src/types.rs (hunk 2 above)
Comment (cubisttriangle) [reply src/types.rs:304]: It's possible that something like this already exists, but I didn't look too hard. Mabye [strum](https://crates.io/crates/strum).

**Pull request merged**